telegram/utils/helpers.py

```diff
@@ -251,6 +250,41 @@ def effective_message_type(entity):
     return None
 
 
+def extract_urls(message):
+    """
+    Extracts all Hyperlinks that are contained in a message. This includes message entities and the
+    media caption, i.e. while of course only text *or* caption is present this works for both.
+    Distinct links are returned in order of appearance.
+
+    Note:
+        For exact duplicates, only the first appearence will be kept, but there may still be URLs
+        that link to the same resource.
+
+    Args:
+        message (:obj:`telegram.Message`) The message to extract from
+
+    Returns:
+        :obj:`list`: A list of URLs contained in the message
+    """
+    from telegram import MessageEntity
+
+    types = [MessageEntity.URL, MessageEntity.TEXT_LINK]
+    results = message.parse_entities(types=types)
+    results.update(message.parse_caption_entities(types=types))
+
+    # Sort results by order of appearence, i.e. the MessageEntity offset
+    sorted_results = sorted(results.items(), key=lambda e: e[0].offset)
```

Comment:
First remove duplicates, then sort.
It will be more efficient and we wouldn't need OrderedDict